### PR TITLE
fix: MergeEffects cannot return right value, fix #79

### LINKED ISSUE #79: Policy not matching
Hi guys,

version: `casbin/casbin v3.3.0`

here the conf and the policy I am trying to use:

```conf
[request_definition]
r = sub, dom, act

[policy_definition]
p = sub, dom, act, eft

[role_definition]
g = _, _, _

[policy_effect]
e = some(where (p.eft == allow)) && !some(where (p.eft == deny))

[matchers]
m = g(r.sub, p.sub, r.dom) && r.dom == p.dom && r.act == p.act
```

```
p, admin_domain1, domain1, read, allow
p, admin_domain1, domain1, write, allow
p, admin_domain2, domain2, read, allow
p, admin_domain2, domain2, write, allow
g, alice, admin_domain1, domain1
g, bob, admin_domain2, domain2
```

Now, when I want to test:

```php
        $this->assertTrue($enforcer->enforce('alice', 'domain1', 'read')); // <--- pass
        $this->assertTrue($enforcer->enforce('alice', 'domain1', 'write')); // <--- pass

        $this->assertFalse($enforcer->enforce('alice', 'domain2', 'read')); // <--- fail
        $this->assertFalse($enforcer->enforce('alice', 'domain2', 'write')); // <--- fail

        $this->assertFalse($enforcer->enforce('bob', 'domain1', 'read')); // <--- fail
        $this->assertFalse($enforcer->enforce('bob', 'domain1', 'write')); // <--- fail

        $this->assertTrue($enforcer->enforce('bob', 'domain2', 'read')); // <--- pass
        $this->assertTrue($enforcer->enforce('bob', 'domain2', 'write')); // <--- pass
```

What is wrong with my workflow? Since `alice` is not related at all with `domain2`, the assert should return false, but here, it returns true.

I tried also with the following matcher:
```m = r.sub == p.sub && r.dom == p.dom && r.act == p.act```
but same result

 Thank you for your help.

### PR DESCRIPTION
**Pull request by basakest**
fix: #79

### CODE REVIEW

**Comment by hsluoyz** [comment]
@mmehira plz review.

**Comment by mmehira** [comment]
@hsluoyz just tested locally with the same conf and policy files. Everything is working as expected. 

Good job! Thanks a lot for this fix.

**Comment by hsluoyz** [comment]
@techoner

File: src/Effect/DefaultEffector.php [hunk 1]
```diff
@@ -40,10 +40,14 @@ public function mergeEffects(string $expr, array $effects, array $results): arra
...
+
             $explainIndex = array_search(self::DENY, $effects, true);
-            if ($explainIndex !== false) {
-                $result = false;
+            if ($explainIndex === false) {
+                $temp = array_search(self::ALLOW, $effects, true);
+                if ($temp !== false) {
+                    $result = true;
+                }
             }
```
Comment (techoner) [inline src/Effect/DefaultEffector.php:51]: @basakest This can be a `for` loop, just one loop, avoid using `array_search` twice .

**Review by techoner — CHANGES_REQUESTED** [review]

**Review by techoner — APPROVED** [review]

**Pull request merged**